src/PhpPact/Consumer/Driver/Interaction/InteractionDriverInterface.php

```diff
@@ -8,7 +8,7 @@ interface InteractionDriverInterface extends DriverInterface
 {
     public function uponReceiving(string $description): void;
 
-    public function withBody(string $part, ?string $contentType = null, ?string $body = null): void;
+    public function withBody(bool $isRequest, ?string $contentType = null, ?string $body = null): void;
```

Comment:
Instead of using feature flags, `InteractionDriverInterface` can be implemented twice separately for Request and Response with common parts shared via trait/abstract class. It will simplify usage and readability.

Reply:
Thanks for your valuable time reviewing such a big PR like this and giving back great reviews. I split `InteractionDriver` into smaller classes. Hope it is aligned with your suggestion.